### PR TITLE
Master stock dropshipping new api vgo

### PR DESCRIPTION
**Pull request by vgo-dev**
Task : https://www.odoo.com/web#id=12979&view_type=form&model=project.task&menu_id=3942&action=327

Pad: https://pad.odoo.com/p/r.g4GVWnV0lMAveY36

### CODE REVIEW

File: addons/stock_dropshipping/tests/common.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        self.partner_sup_id = self.env.ref('base.res_partner_2').id
+        self.product_uom_unit_id = self.env.ref('product.product_uom_unit').id
+        self.product_uom_dozen_id = self.env.ref('product.product_uom_dozen').id
+        self.product_uom_kgm_id = self.env.ref('product.product_uom_kgm').id
+        self.warehouse_id = self.env.ref('stock.warehouse0').id
+        self.account_payment_term_id = self.env.ref('account.account_payment_term').id
+        self.route_drop_shipping_id = self.env.ref('stock_dropshipping.route_drop_shipping').id
+        self.account_type_income_id = self.env.ref('account.data_account_type_other_income').id
+        self.account_type_expense_id = self.env.ref('account.data_account_type_expenses').id
+        self.stock_customers_location_id = self.env.ref('stock.stock_location_customers').id
```
Comment (WHPODOO) [inline addons/stock_dropshipping/tests/common.py]: if you use reference ***id*** then no need to use ***env***
```
varable_nem_id = self.ref('<reference name>')
```

**Comment by WHPODOO** [comment]
In test case you use reference ***id*** of any reference then no need to use ***env***
```
varable_nem_id = self.ref('<reference name>')
```

**Comment by vgo-odoo** [comment]
@WHPODOO  :+1: ,
Its changed.

File: addons/stock_dropshipping/tests/test_lifo_price.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            'property_stock_account_output_categ_id': self.o_expense,
+            'property_account_income_categ_id': self.o_income,
+            'property_stock_account_input_categ_id': self.default_account,
+            'property_stock_journal': self.stock_journal,
+            'property_account_expense_categ_id': self.default_account})
+        # Set a product as using lifo price
+        self.product_lifo_icecream = self.Product.create({'default_code': 'LIFO', 'name': 'LIFO Ice Cream', 'type': 'product', 'categ_id': product_category_001.id, 'list_price': 100.0, 'standard_price': 70.0, 'uom_id': self.product_uom_kgm_id, 'uom_po_id': self.product_uom_kgm_id, 'valuation': 'real_time', 'cost_method': 'real', 'property_stock_account_input': self.o_expense, 'property_stock_account_output': self.o_income, 'description': 'LIFO Ice Cream can be mass-produced and thus is widely available in developed parts of the world. Ice cream can be purchased in large cartons (vats and squrounds) from supermarkets and grocery stores, in smaller quantities from ice cream shops, convenience stores, and milk bars, and in individual servings from small carts or vans at public events.'})
```
Comment (mart-e) [inline addons/stock_dropshipping/tests/test_lifo_price.py]: way too long line

File: addons/stock_dropshipping/tests/common.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        self.route_drop_shipping_id = self.ref('stock_dropshipping.route_drop_shipping')
+        # Create a warehouse with pick-pack-ship and receipt in 2 steps
+        self.wh_pps = self.StockWarehouse.create({'name': 'WareHouse PickPackShip', 'code': 'whpps', 'reception_steps': 'two_steps', 'delivery_steps': 'pick_pack_ship'})
+        # Create Account
+        self.o_income = self.Account.create({'name': 'Opening Income', 'code': 'X1016', 'user_type_id': self.env['account.account.type'].create({'name': 'Other Income', 'type': 'other'}).id})
+        self.o_expense = self.Account.create({'name': 'Opening Expense', 'code': 'X1114', 'user_type_id': self.env['account.account.type'].create({'name': 'Other Expenses', 'type': 'other'}).id})
+        self.o_valuation = self.Account.create({'name': 'Stock Valuation', 'code': 'X1115', 'user_type_id': self.env['account.account.type'].create({'name': 'Stock Valuation', 'type': 'other'}).id})
+        self.default_account = self.Account.create({'name': "Purchased Stocks", 'code': "X1101", 'user_type_id': self.env['account.account.type'].create({'name': 'Expenses', 'type': 'other'}).id, 'reconcile': True})
```
Comment (mart-e) [inline addons/stock_dropshipping/tests/common.py]: are you sure, this part doesn't look to be present in the original tests

**Comment by mart-e** [comment]
Hello,

It looks like this code is from the old accounting, the tests are totally different. Please when rebasing your code, you **must** adapt your changes.

File: addons/stock_dropshipping/tests/common.py (hunk 3 above)
Comment (vgo-odoo) [inline addons/stock_dropshipping/tests/common.py]: @mart-e   Here https://github.com/odoo/odoo/blob/master/addons/account/test/account_minimal_test.xml#L29 reference file which is load the accounts but in test case need only 4 account.
so, do you think it is better load the whole data from <account_minimal> file than creating only 4 records that we need in our test cases ?

File: addons/stock_dropshipping/__openerp__.py [hunk 4]
```diff
@@ -22,7 +22,7 @@
 """,
     'website': 'https://www.odoo.com/page/warehouse',
     'depends': ['purchase', 'sale_stock'],
-    'data': ['stock_dropshipping.xml'],
+    'data': ['data/stock_dropshipping.xml'],
```
Comment (tde-banana-odoo) [inline addons/stock_dropshipping/__openerp__.py]: Rename to stock_dropshipping_data.xml

File: addons/stock_dropshipping/data/stock_dropshipping.xml [hunk 5]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0"?>
-<openerp>
-    <data>
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (tde-banana-odoo) [inline addons/stock_dropshipping/data/stock_dropshipping.xml]: This diff has nothing to do with the commit.

File: addons/stock_dropshipping/tests/common.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+from odoo.tests import common
+
+
+class TestStockDropshippingCommon(common.TransactionCase):
+
+    def setUp(self):
+        super(TestStockDropshippingCommon, self).setUp()
+
+        self.ProcurementGroup = self.env['procurement.group']
```
Comment (tde-banana-odoo) [inline addons/stock_dropshipping/tests/common.py]: Too much self.something. I bet most of them are unnecessary?

**Comment by tde-banana-odoo** [comment]
Task cancelled.

**Pull request closed without merging**